include/envoy/server/filter_config.h

```diff
@@ -200,6 +200,21 @@ class FactoryContext : public virtual CommonFactoryContext {
   virtual ProtobufMessage::ValidationVisitor& messageValidationVisitor() PURE;
 };
 
+/**
+ * An implementation of FactoryContext. It should be used to create NetworkFilterChain.
+ */
+class FilterChainFactoryContext : public virtual FactoryContext {
+public:
+  /**
+   * @return uint64_t The tag assigned with this context at runtime. It will be used for connection
+   * handler tracking.
+   */
+  virtual uint64_t getTag() const PURE;
```

Comment:
I get why there is a tag on the filter chain, but not on the factory context. Which tag is this? Is this now the listener tag? Maybe given listener and filter chain tags different names, e.g. `filter_chain_tag`, `listener_tag`, to make this relationship clear.

Reply:
Will rename in the next commit. It's still filter_chain tag here

Reply:
already killed T_T